### ISSUE TITLE
Zero Division in Preprocess Spectra

### ISSUE THREAD

**Issue by Guillaumes33** [Quasars/orange-spectroscopy#345]
Setting Gaussian smoothing to 0 gives a zero division error. Not really unexpected, but risky when it brings up the option to report the error, and reporting it will close Orange without saving.

**Comment by borondics** [comment]
Good catch @Guillaumes33 !

The SciPy function we use doesn't handle 0...
```
from scipy.ndimage import gaussian_filter1d 
gaussian_filter1d([1.0, 2.0, 3.0, 4.0, 5.0], 0) 
```
@markotoplak @stuart-cls an easy fix would be to limit the "bottom" property of the textbox to some small value, but that is an ugly hack. Any better ideas?

**Comment by borondics** [comment]
Now that I think about this, there is another problem with the upper limit. 

Currently, the textbox input is limited to 0-1000. What if someone loads a hard x-ray spectrum with several tens of thousand eV energy range? 

The allowed upper limit should be calculated from the energy range of the data.

**Comment by markotoplak** [comment]
In Guassian smoothing the parameter is connected to number of neighbor values and not with the actual insterval.  Like Savitzky-golay filter window size.

**Comment by Guillaumes33** [comment]
I don't really know enough to tell if this is a valid assumption, but it seems like SD = 0 could be equivalent to no smoothing at all. Could you set it so that the input spectra are unaltered when SD = 0, or does that not fit with what more experienced people would expect?

**Comment by stuart-cls** [comment]
Fixed in #348
